x-pack/plugins/lens/public/datatable_visualization/components/columns.tsx

```diff
@@ -151,31 +151,33 @@ export const createGridColumns = (
 
     const additionalActions: EuiListGroupItemProps[] = [];
 
-    if (!isReadOnly) {
+    additionalActions.push({
+      color: 'text',
+      size: 'xs',
+      onClick: () => onColumnResize({ columnId: originalColumnId || field, width: undefined }),
+      iconType: 'empty',
+      label: i18n.translate('xpack.lens.table.resize.reset', {
+        defaultMessage: 'Reset width',
+      }),
+      'data-test-subj': 'lensDatatableResetWidth',
+      isDisabled: initialWidth == null,
+    });
+    if (!isTransposed) {
       additionalActions.push({
         color: 'text',
         size: 'xs',
-        onClick: () => onColumnResize({ columnId: originalColumnId || field, width: undefined }),
-        iconType: 'empty',
-        label: i18n.translate('xpack.lens.table.resize.reset', {
-          defaultMessage: 'Reset width',
+        onClick: () => onColumnHide({ columnId: originalColumnId || field }),
```

Comment:
Can we make sure that focus isn't lost after this? (Right now it returns to the `<body>`?

Ideally would maybe be the first cell of the grid but I'm not sure if that's supported by the Grid component. Otherwise, focusing the grid and whatever happens is probably fine.

Reply:
Hi @myasonik , thanks for this comment. As it's not related to the PR and an issue in a released version I would like to split out this part. I opened an issue herre: https://github.com/elastic/kibana/issues/102156